core/src/main/java/io/questdb/std/bytes/DirectByteSink.java

```diff
@@ -93,6 +93,20 @@ public void advance(long written) {
         setImplPtr(getImplPtr() + written);
     }
 
+    @Override
+    public @NotNull NativeByteSink borrowDirectByteSink() {
+        if (impl == 0) {
+            throw new IllegalStateException("Sink is closed");
```

Comment:
this should be an assert, we do not handle IllegalStateExceptions at runtime, the common pattern thougout the code is to use `assert`

Reply:
Done.